src/js/modules/articles/actions.js

```diff
@@ -1,11 +1,67 @@
 import axios from 'axios';
 import * as t from './actionTypes';
+import { getSections } from '../sections/selectors';
 
-const apiUrl = 'http://api.icndb.com/jokes/random';
+const apiUrl = 'https://api.stuyspec.xyz/articles';
 
+//TODO: need a way to call the fetch function
 export const fetch = () => {
-	return {
-		type: t.FETCH_ARTICLE,
-		payload: axios.get(apiUrl),
-	};
+  return (dispatch, getState) => {
+    dispatch({ type: t.FETCH_ARTICLE_PENDING });
+    //axios.get(apiUrl, {'headers': {'X-Key-Inflection': 'camel'}}) TODO: wait until Nic uses olive branch
+    axios.get(apiUrl)
+      .then((info) => {
+        // This current code will dispatch the FETCH_ARTICLE_REJECtED because the API still has null values
+        if (articleValidator(info.data)) {
+          const sections = getSections(getState());
+          dispatch({
+            type: t.FETCH_ARTICLE_FULFILLED,
+            payload: info.data,
+            sections: sections,
+          })
+        } else {
+          dispatch({ type: t.FETCH_ARTICLE_REJECTED, payload: 'Invalid article array' })
+        }
+      })
+      .catch((err) => {
+        dispatch({ type: t.FETCH_ARTICLE_REJECTED, payload: err })
+      })
+  };
 };
+const keyValidator = (articleObject, key, type) => {
+  if (key in articleObject) {
+    return (typeof (articleObject[ key ]) === type );
+  } else {
+    return false;
+  }

```

Comment:
Actually, this can all be condensed into one line:
`return (key in articleObject) && (typeof (articleObjecty[ key ]) === type );`

This means that you can change up the ES6 syntax and omit the return statement:
`const keyValidator = (articleObject, key, type) => (key in articleObject) && (typeof (articleObjecty[ key ]) === type )`. Of course, this statement is a long line and a little unreadable, so you should write line breaks <a href="https://contribute.jquery.org/style-guide/js/#multi-line-statements">wherever appropriate</a>.

After fixing that^ stuff, look through <a href="https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions">this MDN documentation</a>. It explains the different types of arrow functions. That is why, if you look at any `mapStateToProps`, it returns `({ ... })`.